scripts/docs/snippetadapter.js

```diff
@@ -7,83 +7,242 @@
+					if ( snippetData.requiredFor ) {
+						let htmlFile = fs.readFileSync( snippetData.snippetSources.html ).toString();
+
+						if ( wasCSSGenerated ) {
+							htmlFile += '<link rel="stylesheet" href="snippet.css" type="text/css">';
+						}
+
+						htmlFile += '<script src="snippet.js"></script>';
 
-				if ( wasCSSGenerated ) {
-					htmlFile += '<link rel="stylesheet" href="snippet.css" type="text/css">';
+						fs.writeFileSync( path.join( snippetData.outputPath, snippetData.snippetName, 'snippet.html' ), htmlFile );
+
+						continue;
+					}
+
+					let snippetHTML = fs.readFileSync( snippetData.snippetSources.html ).toString();
+
+					snippetHTML = snippetHTML.replace( /%BASE_PATH%/g, snippetData.basePath );
+					snippetHTML = `<div class="live-snippet">${ snippetHTML }</div>`;
+
+					content = content.replace( getSnippetPlaceholder( snippetData.snippetName ), snippetHTML );
+
+					jsFiles.push( path.join( snippetData.basePath, 'assets', 'snippet.js' ) );
+					jsFiles.push( path.join( snippetData.relativeOutputPath, snippetData.snippetName, 'snippet.js' ) );
+
+					cssFiles.push( path.join( snippetData.basePath, 'assets', 'snippet-styles.css' ) );
+
+					if ( wasCSSGenerated ) {
+						cssFiles.unshift( path.join( snippetData.relativeOutputPath, snippetData.snippetName, 'snippet.css' ) );
+					}
 				}
 
-				htmlFile += '<script src="snippet.js"></script>';
+				const cssImportsHTML = getHTMLImports( cssFiles, importPath => {
+					return `    <link rel="stylesheet" href="${ importPath }" type="text/css">`;
+				} );
 
-				fs.writeFileSync( path.join( outputPath, 'snippet.html' ), htmlFile );
-			}
+				const jsImportsHTML = getHTMLImports( jsFiles, importPath => {
+					return `    <script src="${ importPath }"></script>`;
+				} );
 
-			return {
-				html: fs.readFileSync( data.snippetSource.html ),
-				assets: {
-					js: [
-						// Load snippet helpers first.
-						path.join( data.basePath, 'assets', 'snippet.js' ),
-
-						// Then load the actual snippet code.
-						path.join( data.relativeOutputPath, data.snippetPath, 'snippet.js' )
-					],
-					css: cssFiles
-				},
-				dependencies: snippetConfig.dependencies
-			};
+				content = content.replace( '<!--UMBERTO: SNIPPET: CSS-->', cssImportsHTML );
+				content = content.replace( '<!--UMBERTO: SNIPPET: JS-->', jsImportsHTML );
+
+				fs.writeFileSync( destinationPath, content );
+			}
+		} )
+		.then( () => {
+			console.log( 'Snippets has been built.' );
 		} );
 };
 
-function getWebpackConfig( config ) {
+/**
+ * Removes snippets that names do not match to patterns specified in `whitelistedSnippets` array.
+ *
+ * @param {Set.<Snippet>} snippets Snippet collection extracted from documentation files.
+ * @param {Array.<String>} whitelistedSnippets Snippet patterns that should be built.
+ */
+function filterWhitelistedSnippets( snippets, whitelistedSnippets ) {
+	if ( !whitelistedSnippets.length ) {
```

Comment:
whitelistedSnippets might be not defined ( this happen in big bang docs ), you should handle such case, and build all snippets in such scenario.
```
Executing the snippet adapter for "ckeditor5".
TypeError: Cannot read property 'length' of undefined
    at filterWhitelistedSnippets (/bigbang-docs.ckeditor.com/projects/ckeditor5/scripts/docs/snippetadapter.js:200:28)
    at snippetAdapter (/bigbang-docs.ckeditor.com/projects/ckeditor5/scripts/docs/snippetadapter.js:78:2)
    at promise.then (/umberto/scripts/filter/after-post-render/snippets.js:48:12)
```

Reply:
Fixed.

Reply:
Now the function will not be called if `whitelistedSnippets` is `undefined`.